cansat-stm32f446/src/main.rs

```diff
@@ -144,21 +218,48 @@ mod app {
 
     #[task(local = [delay, bme])]
     fn bme_measure(ctx: bme_measure::Context) {
-        let bme = ctx.local.bme;
-        let delay = ctx.local.delay;
-        let measurements = match bme.measure(delay) {
-            Ok(m) => m,
-            Err(e) => {
-                defmt::error!("Could not read bme280 measurements: {}", Debug2Format(&e));
-                return;
-            }
+        #[cfg(feature = "bme")]
+        {
+            let bme = ctx.local.bme;
+            let delay = ctx.local.delay;
+            let measurements = match bme.measure(delay) {
+                Ok(m) => m,
+                Err(e) => {
+                    defmt::error!("Could not read bme280 measurements: {}", Debug2Format(&e));
+                    return;
+                }
+            };
+            let altitude = cansat::calculate_altitude(measurements.pressure);
+            defmt::info!("Altitude = {} meters above sea level", altitude);
+            defmt::info!("Relative Humidity = {}%", measurements.humidity);
+            defmt::info!("Temperature = {} deg C", measurements.temperature);
+            defmt::info!("Pressure = {} pascals", measurements.pressure);
+
+            bme_measure::spawn_after(5.secs()).unwrap();
+        }
+    }
+    #[task(local = [controller, filename])]
+    fn sdmmc_log(ctx: sdmmc_log::Context, log_string: String<11>) {
+        let controller = ctx.local.controller;
+        let filename = ctx.local.filename;
+        let mut volume = match controller.get_volume(embedded_sdmmc::VolumeIdx(0)) {
+            Ok(volume) => volume,
+            Err(e) => defmt::panic!(":) {}", e),
         };
-        let altitude = cansat::calculate_altitude(measurements.pressure);
-        defmt::info!("Altitude = {} meters above sea level", altitude);
-        defmt::info!("Relative Humidity = {}%", measurements.humidity);
-        defmt::info!("Temperature = {} deg C", measurements.temperature);
-        defmt::info!("Pressure = {} pascals", measurements.pressure);
 
-        bme_measure::spawn_after(5.secs()).unwrap();
+        let root_dir = controller.open_root_dir(&volume).unwrap();
+
+        let mut f = controller
```

Comment:
You didn't close the file.

I wonder, do we have to close the file after every message to flush the buffer? Maciek told me that its what they had to do in OBC:
```c++
void file_appendln(const char* file_name, const char* data)
{
    // NOLINTNEXTLINE (hicpp-signed-bitwise)
    file = SD.open(file_name, O_CREAT | O_WRITE | O_APPEND);
    file.println(data);
    file.close();
}
```

We will need to do some testing, but we can do this after we merge this PR.